### PR TITLE
Feature - Add final process submission page and functionality

### PR DESCRIPTION
**Pull request by jonathangoulding**

### CODE REVIEW

File: packages/forms-web-app/__tests__/unit/controllers/examination/process-submission/utils/mock-file-2.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jonathangoulding) [inline packages/forms-web-app/__tests__/unit/controllers/examination/process-submission/utils/mock-file-2.json:1]: maybe put these into a mock files folder

File: packages/forms-web-app/src/controllers/examination/middleware/submission.middleware.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			directory,
+			pages: { submissionComplete }
+		}
+	}
+} = require('../../../routes/config');
+
+const isProcessingSubmission = (req, res, next) => {
+	const { url, session } = req;
+
+	if (url === submissionComplete.route) return next();
```
Comment (jonathangoulding) [inline packages/forms-web-app/src/controllers/examination/middleware/submission.middleware.js:18]: would be nice not have this. But seems required to lock all the other routes out and allow the submission complete to be shown. 

We can flip the logic below so if no condition next etc.

File: packages/forms-web-app/src/controllers/examination/session/examination-session.js [hunk 3]
```diff
@@ -1,9 +1,46 @@
...
 	const examinationSession = session?.examination;
 	if (!examinationSession) throw new Error('No examination session');
 	return examinationSession;
 };
 
+const setExaminationUploadingState = (session, state) => {
+	const examinationSession = getExaminationSession(session);
+	if (typeof state !== 'boolean') throw new Error('Examination upload state is not a boolean');
+	examinationSession.uploading = state;
+	// Save is necessary to save the session in between states- https://stackoverflow.com/a/26532987
```
Comment (jonathangoulding) [inline packages/forms-web-app/src/controllers/examination/session/examination-session.js:13]: comment feels necessary.

File: packages/forms-web-app/src/services/submission.service.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const fetch = require('node-fetch');
+const logger = require('../lib/logger');
+const config = require('../config');
+
+const postSubmission = async (caseRef, body) => {
+	try {
+		const url = `${config.applications.url}/api/v1/submissions/${caseRef}`;
```
Comment (jonathangoulding) [inline packages/forms-web-app/src/services/submission.service.js:7]: URL may need work to get it from config ?

Comment (Philip-Philippou1) [reply packages/forms-web-app/src/services/submission.service.js:7]: All other API calls are going through the wrapper - application-api-wrapper, which handles the url and catches error responses.  Is it worth adding this there?

File: packages/forms-web-app/src/controllers/examination/process-submission/utils/helpers.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	const titleSplit = title.split('-');
+	return titleSplit[1].trim();
+};
+
+const mapSessionToCommonFormData = (examination, item) => {
+	const { hasInterestedPartyNo, name, email, title } = examination;
+	const formData = new FormData();
+	formData.append('name', name);
+	formData.append('email', email);
+	formData.append('interestedParty', `${hasInterestedPartyNo === 'yes'}`);
```
Comment (Philip-Philippou1) [inline packages/forms-web-app/src/controllers/examination/process-submission/utils/helpers.js:12]: Do we have interested party number  ?

File: packages/forms-web-app/src/controllers/examination/session/examination-session.js (hunk 3 above)
Comment (Philip-Philippou1) [reply packages/forms-web-app/src/controllers/examination/session/examination-session.js:13]: I would leave it as we will have forgotten in 2 weeks time

File: packages/forms-web-app/src/services/submission.service.js (hunk 4 above)
Comment (Philip-Philippou1) [reply packages/forms-web-app/src/services/submission.service.js:7]: There seems to be a headers argument in the handler, if it's real hassle it's no big one just be nice to have it consistent.

Comment (jonathangoulding) [reply packages/forms-web-app/src/services/submission.service.js:7]: the arg does not change the content type but i can change that as well. 


Will do this.

File: packages/forms-web-app/__tests__/unit/controllers/examination/process-submission/utils/mock-file-2.json (hunk 1 above)
Comment (jonathangoulding) [reply packages/forms-web-app/__tests__/unit/controllers/examination/process-submission/utils/mock-file-2.json:1]: sure

File: packages/forms-web-app/src/controllers/examination/process-submission/utils/helpers.js (hunk 5 above)
Comment (jonathangoulding) [reply packages/forms-web-app/src/controllers/examination/process-submission/utils/helpers.js:12]: added.

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**